### ISSUE TITLE
Mark Lidi as providing type information, please

### ISSUE THREAD

**Issue by aardjon** [AlTosterino/Lidi#7]
Hi, when using `Lidi` (version 0.3.0) in a type-checked project, `mypy` doesn't recognize the provided type annotations because there is no marker file. Can you please provide one?

### Rationale

Minimal example:
```python
# To setup, run: pip install mypy lidipy
from lidipy import Lidi

class SomeInterface:
    pass

lidi_instance = Lidi()
lidi_instance.resolve(SomeInterface, 42)  # I'd like a MyPy warning here
```

Output of running MyPy:
```
$ mypy exymple.py 
example.py:2: error: Skipping analyzing "lidipy": module is installed, but missing library stubs or py.typed marker  [import-untyped]
example.py:2: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)
```

This is because `mypy` [ignores the existing annotations unless the package is explicitly marked as providing type annotations](https://stackoverflow.com/questions/60716482/error-skipping-analyzing-flask-mysqldb-found-module-but-no-type-hints-or-lib).

### How to support type checking
To fix, simply put an empty `py.types` file into the `src/lidipy/` directory, as described [here](https://typing.readthedocs.io/en/latest/spec/distributing.html#packaging-typed-libraries) and [here](https://typing.readthedocs.io/en/latest/guides/libraries.html#providing-type-annotations). MyPy then correctly warns about the unexpected parameter:

```
$ mypy example.py 
example.py:8: error: Too many arguments for "resolve" of "Lidi"  [call-arg]
Found 1 error in 1 file (checked 1 source file)
```

The `py.typed` must be included in the released package, which poetry seems to do automatically.

**Comment by AlTosterino** [comment]
Hi,

Thanks for catching that, already fixed in `0.3.1` 😃. 

I checked your example, and it runs as expected:
```shell
$ pip freeze && mypy main.py                                                                                                                                       ─╯
lidipy==0.3.1
mypy==1.13.0
mypy-extensions==1.0.0
typing_extensions==4.12.2
main.py:10: error: Too many arguments for "resolve" of "Lidi"  [call-arg]
Found 1 error in 1 file (checked 1 source file)
```

**Comment by aardjon** [comment]
Wow, that was really fast! :+1: Thank you very much :bow:
